src/components/dateTimePicker/index.js

```diff
@@ -133,14 +141,14 @@ class DateTimePicker extends BaseComponent {
         // since handleChange() is not called on iOS when there is no actual change
         this.chosenDate = new Date();
       }
-      
+
       _.invoke(this.props, 'onChange', this.chosenDate);
       this.setState({value: this.chosenDate});
     });
 
   getStringValue = () => {
     const {value} = this.state;
```

Comment:
If the component is controlled, the value should derived directly from props and not the state.

Reply:
You're right, generally speaking, but when we want a controlled component and still need to keep an internal state that's the way to do it. See Carousel, DialogDismissibleView, PageControl, Picker, GradientSlider for example.